addons/base_vat/models/res_partner.py

```diff
@@ -463,21 +463,25 @@ def check_vat_ar(self, vat):
     def default_compact(self, vat):
         return vat
 
-    def _fix_vat_number(self, vat):
+    def _fix_vat_number(self, vat, code):
         vat_country, vat_number = self._split_vat(vat)
+        if vat_country != code.lower():
```

Comment:
I think that won't work with Monaco, which uses the French format (but a different country code).